### ISSUE TITLE
Paraview Problems

### ISSUE THREAD

**Issue by jrood-nrel** [LLNL/spack#5735]
I've been using the issues here to solve some of my own problems with applications not installing, so I'm just going to document some issues I had with installing Paraview that I solved, but haven't decided on the best way to put back into Spack.

When installing Paraview I found that it downloads a tar file like normal which has a `version.txt` file in it which contains `5.4.1` for example. The cmake configure then does a `find_package(git)` and if the Git executable exists, it runs `git describe` and uses whatever this output may be, which in my case was `0.8.17-6535-g5ccc9c1` for whatever reason:

```
-- Found Git: /usr/bin/git (found version "1.7.1") 
-- Determined Source Version : 0.8.17-6535-g5ccc9c1
```

Then Paraview is happy to consider itself version 0.8 and when trying to connect with a 5.4.1 client, it fails. The solution for me was to stop Paraview from running `git describe` by commenting it out in the CMakeLists.txt and then it defaults to using `version.txt`:

```
-- Could not use git to determine source version, using version 5.4.1
```

Which then worked fine.

Next the Paraview library files are then installed into `lib/paraview-5.4` rather than the normal `lib` directory itself, so I had to modify the module file that was generated.

I would like to put this back into Spack, but I'm not sure of the best way to deal with the `git describe` issue, by either a patch, or whatever since I'm not sure if my patch will only work for 5.4.x. I feel like it might almost be best to hide `git` from Paraview before configure somehow to solve the problem for all Paraview versions. I will submit this issue to Paraview.

The module file should be easy to generate the correct library path, but I haven't looked into how to customize module file generation yet.

Anyway just documenting my experience in case anyone else deals with Paraview. I will try to get my modifications put back into Spack in the right manner if I find the time.

**Comment by davydden** [comment]
> commenting it out in the CMakeLists.txt and then it defaults to using version.txt:

i think that shall go into a patch, you can make it conditional to the range of version you know it works with. I would still submit the issue upstream

> installed into lib/paraview-5.4

what if you add `-DCMAKE_INSTALL_LIBDIR=self.prefix.lib`? That should works CMake to install into `prefix/lib`.

**Comment by jrood-nrel** [comment]
Thanks. I made a pull request to patch Paraview for this.

I haven't addressed the module file generation yet though. Unfortunately `-DCMAKE_INSTALL_LIBDIR=self.prefix.lib` didn't stop Paraview from installing to `lib/paraview-5.4`.

**Comment by jrood-nrel** [comment]
I submitted this to the Paraview issue tracker: https://gitlab.kitware.com/paraview/paraview/issues/17761

**Comment by jrood-nrel** [comment]
It looks like Paraview has addressed this bug directly and #5783 is probably fine going forward. Once #5799 is merged, I think these issues are resolved so I'm going to close this.
